delivery_tnt_oca/models/tnt_request.py

```diff
@@ -82,20 +84,32 @@ def _prepare_account(self, partner):
 
     def _prepare_rate_shipment(self):
         totalWeight = 0
-        totalVolume = 0
-        for line in self.record.order_line.filtered(
-            lambda x: x.product_id
-            and (x.product_id.weight > 0 or x.product_id.volume > 0)
-        ):
-            totalWeight += line.product_id.weight * line.product_uom_qty
-            totalVolume += line.product_id.volume * line.product_uom_qty
+        lines = self.record.order_line.filtered(
+            lambda x: x.product_id and x.product_id.weight > 0
+        )
+        for line in lines:
+            weight_line = line.product_id.weight
+            # Fix if product_dimension_volumetric_weight addon installed
+            if "volumetric_weight" in line.product_id._fields:
```

Comment:
This is an ugly patch. If TNT works with volumetric height, then the dependency must be explicit.

Reply:
Finally, related changes to `product_dimension_volumetric_weight` has been removed (because it is a deformation of general use case).